metadata-io/src/main/java/com/linkedin/metadata/search/cache/CacheableSearcher.java

```diff
@@ -88,16 +91,19 @@ private SearchResult getBatch(int batchId) {
       QueryPagination batch = getBatchQuerySize(batchId);
       SearchResult result;
       if (enableCache()) {
-        Timer.Context cacheAccess = MetricUtils.timer(this.getClass(), "getBatch_cache_access").time();
-        K cacheKey = cacheKeyGenerator.apply(batch);
-        result = cache.get(cacheKey, SearchResult.class);
-        cacheAccess.stop();
-        if (result == null) {
-          Timer.Context cacheMiss = MetricUtils.timer(this.getClass(), "getBatch_cache_miss").time();
-          result = searcher.apply(batch);
-          cache.put(cacheKey, result);
-          cacheMiss.stop();
-          MetricUtils.counter(this.getClass(), "getBatch_cache_miss_count").inc();
+        try (Timer.Context ignored2 = MetricUtils.timer(this.getClass(), "getBatch_cache").time()) {
+          Timer.Context cacheAccess = MetricUtils.timer(this.getClass(), "getBatch_cache_access").time();
+          K cacheKey = cacheKeyGenerator.apply(batch);
+          String json = cache.get(cacheKey, String.class);
+          result = json != null ? toRecordTemplate(SearchResult.class, json) : null;
```

Comment:
Hazelcast requires that the cached object is a string? Or serializable? Cannot record template be serialized by itself?

Reply:
Requires that it be serializable, RecordTemplate does not implement Serializable and any RecordTemplate being used in the key or value was throwing errors. Looked into trying to inject a custom deserializer into the Hazelcast deserialization config, but this was much easier.